packages/react-devtools-shared/src/__tests__/statefulStoreSerializer.js

```diff
@@ -0,0 +1,24 @@
+import {printStore} from 'react-devtools-shared/src/devtools/utils';
+
+// test() is part of Jest's serializer API
+export function test(maybeStoreWithState) {
+  if (maybeStoreWithState === null || typeof maybeStoreWithState !== 'object') {
+    return false;
+  }
+
+  const {store: maybeStore} = maybeStoreWithState;
+
+  // It's important to lazy-require the Store rather than imported at the head of the module.
+  // Because we reset modules between tests, different Store implementations will be used for each test.
+  // Unfortunately Jest does not reset its own serializer modules.
+  return (
+    maybeStore instanceof
+    require('react-devtools-shared/src/devtools/store').default
+  );
+}
```

Comment:
How does this serializer _not_ conflict with `storeSerializer`? And what's the purpose of making a separate serializer? Not clear at all to me from a glance why I'd want to use "stateful store" or "store" serializer. (The store is always _stateful_ anyway.)

Reply:
Nevermind. It's early here and I was totally not seeing this line.
```js
const {store: maybeStore} = maybeStoreWithState;
```

Reply:
I don't like the name of this serializer since "stateful store" means "the store" to me. Let me think about it a little bit though. :)

Reply:
The naming is definitely off. I just chose the name because we need the `state` + `store` hence `statefulStore`. 

The main goal was to not affect existing tests since I didn't know if you would find the new serializer helpful. 

> How does this serializer not conflict with storeSerializer? 

`storeSerializer` just passes a `store` while `statefulStoreSerializer` passes `store` and `state`. The naming causes the confusion here I suppose. That's where the `selectedElementIndex` is coming from. Maybe rename `state` to `treeState` instead?

Feel free to mark this as resolved if 4a8738371577f8d4cb8222c1fe49580f47206bdc addressed the problems for you.

Reply:
Yeah, the comment you're replying to just reflected some initially confusion on my part because I overlooked a line (https://github.com/facebook/react/pull/20463#discussion_r545797719)